lib/snapshot_logger/snapshot_logger.ex

```diff
@@ -0,0 +1,129 @@
+defmodule SnapshotLogger.SnapshotLogger do
+  @moduledoc """
+  Logs algorithm snapshots on a cadence.
+  """
+  use GenServer
+  require Logger
+  alias Alerts.Alert
+  alias AlertsViewer.DelayAlertAlgorithm
+  alias Routes.{Route, RouteStatsPubSub}
+
+  @cadence 5 * 60 * 1000
+
+  @spec start_link(any) :: :ignore | {:error, any} | {:ok, pid}
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts)
+  end
+
+  @impl GenServer
+  def init(opts) do
+    delay_alert_algorithm_modules =
+      Application.get_env(:alerts_viewer, :delay_alert_algorithm_components)
+
+    bus_alerts = Alerts.subscribe() |> filtered_by_bus() |> filtered_by_delay_type()
+    bus_routes = Routes.all_bus_routes(Keyword.get(opts, :route_opts, []))
```

Comment:
```suggestion
    bus_routes = opts |> Keyword.get(:route_opts, []) |> Routes.all_bus_routes()
```
suggestion: Using a pipe chain might help the readability.